@PreAuthorize() does not support space before string expression

**Description**
If you include the spring-security extension and use the functionality with a PreAuthorize annotation for calling a method on a bean, startup fails if a string in the expression has space before an Apostrophe defining a string argument.
The problem is in this line where _argumentExpression_ is used rather than _trimmedArgumentExpression_ when evaluating:
https://github.com/quarkusio/quarkus/blob/12ad774423d09e81731e85a1a3268bd09077829f/extensions/spring-security/deployment/src/main/java/io/quarkus/spring/security/deployment/BeanMethodInvocationGenerator.java#L146

**Expected behavior**
Spaces should be trimmed before evaluating the expression

**Actual behavior**
The string is trimmed, but it is the untrimmed string that is used for evaluation.
Startup fails with the following:
```
Build step io.quarkus.spring.security.deployment.SpringSecurityProcessor#addSpringPreAuthorizeSecurityCheck threw an exception: java.lang.IllegalArgumentException: Expression: '@roleChecker.check('tenant', 'READ')' in the @PreAuthorize annotation on method 'role' of class 'com.acme.SecuredResource' is malformed
        at io.quarkus.spring.security.deployment.SpringSecurityProcessorUtil.createGenericMalformedException(SpringSecurityProcessorUtil.java:22)
        at io.quarkus.spring.security.deployment.BeanMethodInvocationGenerator.generateSecurityCheck(BeanMethodInvocationGenerator.java:194)
        at io.quarkus.spring.security.deployment.SpringSecurityProcessor.addSpringPreAuthorizeSecurityCheck(SpringSecurityProcessor.java:478)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)

        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:181)
        at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:178)
        at io.quarkus.dev.DevModeMain.restartApp(DevModeMain.java:208)
        at io.quarkus.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:137)
        at io.quarkus.vertx.http.deployment.devmode.VertxHotReplacementSetup$1.handle(VertxHotReplacementSetup.java:51)
        at io.quarkus.vertx.http.deployment.devmode.VertxHotReplacementSetup$1.handle(VertxHotReplacementSetup.java:43)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:316)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.spring.security.deployment.SpringSecurityProcessor#addSpringPreAuthorizeSecurityCheck threw an exception: java.lang.IllegalArgumentException: Expression: '@roleChecker.check('tenant', 'READ')' in the @PreAuthorize annotation on method 'role' of class 'com.acme.SecuredResource' is malformed
        at io.quarkus.spring.security.deployment.SpringSecurityProcessorUtil.createGenericMalformedException(SpringSecurityProcessorUtil.java:22)
        at io.quarkus.spring.security.deployment.BeanMethodInvocationGenerator.generateSecurityCheck(BeanMethodInvocationGenerator.java:194)
        at io.quarkus.spring.security.deployment.SpringSecurityProcessor.addSpringPreAuthorizeSecurityCheck(SpringSecurityProcessor.java:478)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)

        at io.quarkus.builder.Execution.run(Execution.java:108)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:128)
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:114)
        ... 10 more
Caused by: java.lang.IllegalArgumentException: Expression: '@roleChecker.check('tenant', 'READ')' in the @PreAuthorize annotation on method 'role' of class 'com.acme.SecuredResource' is malformed
        at io.quarkus.spring.security.deployment.SpringSecurityProcessorUtil.createGenericMalformedException(SpringSecurityProcessorUtil.java:22)
        at io.quarkus.spring.security.deployment.BeanMethodInvocationGenerator.generateSecurityCheck(BeanMethodInvocationGenerator.java:194)
        at io.quarkus.spring.security.deployment.SpringSecurityProcessor.addSpringPreAuthorizeSecurityCheck(SpringSecurityProcessor.java:478)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```

**To Reproduce**
Steps to reproduce the behavior:
1. include spring-security
2. Annotate a resource with PreAuthorize like 
`@PreAuthorize("@roleChecker.check('tenant', 'READ')")`
3. create a bean for evaluating this like:
```
@Component()
public class RoleChecker {
    public boolean check(String tenant, String role) {
        return true;
    }
}
```
4. Start the application and watch the build fail. 

**Environment:**
 - Quarkus version or git rev: 1.2.0.Final

--- a/extensions/spring-security/deployment/src/main/java/io/quarkus/spring/security/deployment/BeanMethodInvocationGenerator.java
+++ b/extensions/spring-security/deployment/src/main/java/io/quarkus/spring/security/deployment/BeanMethodInvocationGenerator.java
@@ -72,7 +72,9 @@ class BeanMethodInvocationGenerator {
         int parametersEndIndex = expression.indexOf(')');
         String[] beanMethodArgumentExpressions = {};
         if (parametersEndIndex > parametersStartIndex + 1) {
-            beanMethodArgumentExpressions = expression.substring(parametersStartIndex + 1, parametersEndIndex).split(",");
+            beanMethodArgumentExpressions = expression.substring(parametersStartIndex + 1, parametersEndIndex).trim()
+                    .split("\\s*,\\s*");
+            ;
         }
         /*
          * We need to make sure the cache key contains the both the expression and the parameter types of the method

--- a/extensions/spring-security/deployment/src/test/java/io/quarkus/spring/security/deployment/BeanMethodCheckTest.java
+++ b/extensions/spring-security/deployment/src/test/java/io/quarkus/spring/security/deployment/BeanMethodCheckTest.java
@@ -86,6 +86,15 @@ public class BeanMethodCheckTest {
     }
 
     @Test
+    public void testWithParamsAndConstant() {
+        assertSuccess(() -> beanWithBeanMethodChecks.withParamAndConstant(new Person("geo")), "withParamAndConstant",
+                ANONYMOUS);
+        assertFailureFor(() -> beanWithBeanMethodChecks.withParamAndConstant(new Person("other")), ForbiddenException.class,
+                USER);
+        assertSuccess(() -> beanWithBeanMethodChecks.withParamAndConstant(new Person("geo")), "withParamAndConstant", USER);
+    }
+
+    @Test
     public void testWithExtraUnusedParam() {
         assertFailureFor(() -> someInterface.doSomething("other", 1, new Person("geo")), UnauthorizedException.class,
                 ANONYMOUS);

--- a/extensions/spring-security/deployment/src/test/java/io/quarkus/spring/security/deployment/springapp/BeanWithBeanMethodChecks.java
+++ b/extensions/spring-security/deployment/src/test/java/io/quarkus/spring/security/deployment/springapp/BeanWithBeanMethodChecks.java
@@ -21,6 +21,11 @@ public class BeanWithBeanMethodChecks {
         return "withParams";
     }
 
+    @PreAuthorize("@personChecker.check(#person, 'geo')")
+    public String withParamAndConstant(Person person) {
+        return "withParamAndConstant";
+    }
+
     @PreAuthorize("@personChecker.check(#person, #input)")
     public String anotherWithParams(String input, Person person) {
         return "anotherWithParams";

